extensions/cache/runtime/src/main/java/io/quarkus/cache/runtime/augmented/AugmentedCacheResult.java

```diff
@@ -0,0 +1,21 @@
+package io.quarkus.cache.runtime.augmented;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import javax.enterprise.util.Nonbinding;
+import javax.interceptor.InterceptorBinding;
+
+@InterceptorBinding
+@Retention(value = RetentionPolicy.RUNTIME)
+@Target({ ElementType.TYPE, ElementType.METHOD })
+public @interface AugmentedCacheResult {
```

Comment:
Just curious, why did you need to go for the `AugmentedCacheInvalidate` trick with annotation replacement?

Reply:
In the current version of the PR, the idea is to keep the API separated from the implementation logic. For example:
- `@CacheResult` is a simple annotation with a mandatory `cacheName` parameter and it can only target methods
- `@AugmentedCacheResult` is an interceptor binding with an optional `cacheName` and it can target both classes and methods

This is only the first version, I need to continue working on the `@CacheOperation` (which will be renamed) annotation I mentioned in another comment here, it will replace all `@AugmentedCache*` annotations.